### PR TITLE
unmanaging items correctly so they are removed from dynamic group

### PR DESCRIPTION
**Pull request by zan-mateusz**

### CODE REVIEW

File: src/main/java/io/cloudsoft/terraform/TerraformConfigurationImpl.java [hunk 1]
```diff
@@ -74,7 +72,22 @@ protected void preStop() {
     @Override
     protected void postStop() {
         getChildren().forEach(c -> c.sensors().set(Attributes.SERVICE_STATE_ACTUAL, Lifecycle.STOPPED));
-        getChildren().forEach(child -> removeChild(child));
+        getChildren().forEach(child -> {
+            if (child instanceof BasicGroup){
+                child.getChildren().forEach(grandChild -> {
+                    if (grandChild instanceof ManagedResource || grandChild instanceof DataResource){
+                        removeChild(grandChild);
+                        Entities.unmanage(grandChild);
```
Comment (algairim) [inline src/main/java/io/cloudsoft/terraform/TerraformConfigurationImpl.java:80]: `unmanage` takes care of removing entity, no?

File: src/main/java/io/cloudsoft/terraform/TerraformConfigurationImpl.java [hunk 2]
```diff
@@ -74,7 +72,22 @@ protected void preStop() {
     @Override
     protected void postStop() {
         getChildren().forEach(c -> c.sensors().set(Attributes.SERVICE_STATE_ACTUAL, Lifecycle.STOPPED));
-        getChildren().forEach(child -> removeChild(child));
+        getChildren().forEach(child -> {
```
Comment (grkvlt) [inline src/main/java/io/cloudsoft/terraform/TerraformConfigurationImpl.java:75]: I think you can trust `unmanage()` to recursively unmanage all children...

File: src/main/java/io/cloudsoft/terraform/TerraformConfigurationImpl.java (hunk 1 above)
Comment (grkvlt) [reply src/main/java/io/cloudsoft/terraform/TerraformConfigurationImpl.java:80]: yes, and does it recursively too...

Comment (grkvlt) [reply src/main/java/io/cloudsoft/terraform/TerraformConfigurationImpl.java:80]: turns out it doesn't, so the remove/unmanage call pair is what is needed

File: src/main/java/io/cloudsoft/terraform/TerraformConfigurationImpl.java (hunk 2 above)
Comment (grkvlt) [reply src/main/java/io/cloudsoft/terraform/TerraformConfigurationImpl.java:75]: nope, you can't

**Review by grkvlt — APPROVED** [review]
Looks good

File: src/main/java/io/cloudsoft/terraform/TerraformConfigurationImpl.java [hunk 3]
```diff
@@ -7,9 +7,7 @@
 import com.google.common.base.Supplier;
 import com.google.common.collect.Maps;
 import com.google.gson.internal.LinkedTreeMap;
-import io.cloudsoft.terraform.entity.DataResource;
-import io.cloudsoft.terraform.entity.ManagedResource;
-import io.cloudsoft.terraform.entity.TerraformResource;
+import io.cloudsoft.terraform.entity.*;
```
Comment (grkvlt) [inline src/main/java/io/cloudsoft/terraform/TerraformConfigurationImpl.java:10]: One thing, can you fix the wildcard import please?

File: src/main/java/io/cloudsoft/terraform/TerraformConfigurationImpl.java [hunk 4]
```diff
@@ -74,7 +72,22 @@ protected void preStop() {
     @Override
     protected void postStop() {
         getChildren().forEach(c -> c.sensors().set(Attributes.SERVICE_STATE_ACTUAL, Lifecycle.STOPPED));
-        getChildren().forEach(child -> removeChild(child));
+        getChildren().forEach(child -> {
+            if (child instanceof BasicGroup){
+                child.getChildren().forEach(grandChild -> {
+                    if (grandChild instanceof ManagedResource || grandChild instanceof DataResource){
```
Comment (grkvlt) [inline src/main/java/io/cloudsoft/terraform/TerraformConfigurationImpl.java:78]: Actually, this can be `child instanceof TerraformResource` instead

**Review by grkvlt — APPROVED** [review]
Thanks

**Pull request merged**